modules/jdbc/src/main/java/org/apache/ignite/jdbc/IgniteJdbcDriver.java

```diff
@@ -29,11 +29,104 @@
 
 /**
- * JDBC driver thin implementation for Apache Ignite 3.x.
+ * JDBC driver implementation for Apache Ignite 3.x.
  *
- * <p>Driver allows to get distributed data from Ignite Data Storage using standard SQL queries and standard JDBC API.
+ * <p>Driver allows to get distributed data from Ignite 3 Data Storage using standard SQL queries and standard JDBC API.</p>
+ * <h2>Register the JDBC drivers</h2>
+ * <p>
+ * The JDBC driver registration is automatically done via the Java Standard Edition Service Provider mechanism.
+ * Ignite JDBC driver implements this feature and it is automatically registered for case the JDBC driver jar presents in classpath.
+ * </p>
+ * <h2>URL Format</h2>
+ * <p>
+ * The JDBC Driver supports the following URL formats to establish a connection with an Ignite 3 Database.
+ * </p>
+ * jdbc:ignite:thin://host[:port][,host[:port][/schema][[?parameter1=value1][&amp;parameter2=value2],...]]
+ * <p> or </p>
+ * jdbc:ignite:thin://host[:port][,host[:port][/schema][[?parameter1=value1][;parameter2=value2],...]]
+ * <p>
+ * URL can have an optional list of name-value pairs as parameters after the '<B>?</B>' delimiter.
+ * Name and value are separated by an '<B>=</B>' and multiple properties are separated either by an '<B>&amp;</B>' or a '<B>;</B>'.
+ * Separate sign can't be mixed and should be either semicolon or ampersand sign.
+ * </p>
+ *   <br>
+ * <table border="1" >
+ *     <caption><b>The list of supported name value pairs:</b></caption>
+ *   <tr>
+ *      <th>Parameter Name</th>
+ *      <th>Description</th>
+ *   </tr>
+ *   <tr>
+ *      <td>schema</td>
+ *      <td>Specifies default schema name.</td>
+ *   </tr>
+ *   <tr>
+ *      <td>queryTimeout</td>
+ *      <td>Number of seconds the driver will wait for a <code>Statement</code> object to execute. Zero means there is no limits.</td>
+ *   </tr>
+ *   <tr>
+ *      <td>connectionTimeout</td>
+ *      <td>Number of milliseconds JDBC client will waits for server to response. Zero means there is no limits."</td>
+ *   </tr>
+ *   <tr>
+ *      <td>reconnectThrottlingPeriod</td>
+ *      <td>Sets the reconnect throttling period, in milliseconds. Zero means there is no limits.</td>
+ *   </tr>
+ *   <tr>
+ *      <td>reconnectThrottlingRetries</td>
+ *      <td>Sets the reconnect throttling retries. Zero means there is no limits.</td>
+ *   </tr>
+ *   <tr>
+ *      <td>trustStorePath</td>
+ *      <td>Path to trust store.</td>
+ *   </tr>
+ *   <tr>
+ *      <td>trustStorePassword</td>
+ *      <td>Trust store password.</td>
+ *   </tr>
+ *   <tr>
+ *      <td>keyStorePath</td>
+ *      <td>Path to key store.</td>
+ *   </tr>
+ *   <tr>
+ *      <td>keyStorePassword</td>
+ *      <td>Key store password.</td>
+ *   </tr>
+ *   <tr>
+ *      <td>clientAuth</td>
+ *      <td>SSL client authentication.</td>
+ *   </tr>
+ *   <tr>
+ *      <td>ciphers</td>
+ *      <td>SSL ciphers.</td>
+ *   </tr>
+ *   <tr>
+ *      <td>sslEnabled</td>
+ *      <td>Enable ssl.</td>
```

Comment:
when do I need to enable SSL? are there parameters that make sense only when `sslEnabled=true`? What about `sslEnabled=false`?

Reply:
fixed